functions/src/events/scrapeEvents.ts

```diff
@@ -138,12 +147,73 @@ class HearingScraper extends EventScraper<HearingListItem, Hearing> {
     return events.filter(HearingListItem.guard)
   }
 
-  async getEvent({ EventId }: HearingListItem) {
-    const content = HearingContent.check(await api.getHearing(EventId))
+  async getEvent({ EventId }: HearingListItem /* e.g. 4962 */) {
+    const data = await api.getHearing(EventId)
+    const content = HearingContent.check(data)
+    const eventInDb = await db
+      .collection("events")
+      .doc(`hearing-${String(EventId)}`)
+      .get()
+    const eventData = eventInDb.data()
+    const hearing = Hearing.check(eventData)
+    const shouldScrape = withinCutoff(hearing.startsAt.toDate())
+
+    let maybeVideoURL = null
+    let transcript = null
+    if (!hearing.videoFetchedAt && shouldScrape) {
+      const req = await fetch(
+        `https://malegislature.gov/Events/Hearings/Detail/${EventId}`
+      )
+      const res = await req.text()
+      if (res) {
+        const dom = new JSDOM(res)
+        if (dom) {
+          const maybeVideoSource =
+            dom.window.document.querySelectorAll("video source")
+          if (maybeVideoSource.length && maybeVideoSource[0]) {
+            const newToken = randomBytes(16).toString("hex")
+            const firstVideoSource = maybeVideoSource[0] as HTMLSourceElement
+            maybeVideoURL = firstVideoSource.src
+
+            transcript = await assembly.transcripts.submit({
+              webhook_url:
+                process.env.NODE_ENV === "development"
+                  ? "https://us-central1-digital-testimony-dev.cloudfunctions.net/transcription"
+                  : "https://us-central1-digital-testimony-prod.cloudfunctions.net/transcription",
+              webhook_auth_header_name: "X-Maple-Webhook",
+              webhook_auth_header_value: newToken,
+              audio: firstVideoSource.src,
+              auto_highlights: true,
+              custom_topics: true,
+              entity_detection: true,
+              iab_categories: false,
+              format_text: true,
+              punctuate: true,
+              speaker_labels: true,
+              summarization: true,
+              summary_model: "informative",
+              summary_type: "bullets"
+            })
+
+            await db
+              .collection("events")
+              .doc(String(EventId))
```

Comment:
does this need to be `hearing-9642` instead of `9642`?

Reply:
Good catch! Fixed in next commit.